scenarios/sns_secrets/terraform/api_gateway.tf

```diff
@@ -0,0 +1,92 @@
+resource "aws_api_gateway_rest_api" "cg_api" {
+  name        = "cg-api-${var.cgid}"
+  description = "API for demonstrating leaked API key scenario"
+  tags = {
+    Name     = "cg-api-${var.cgid}"
+    Stack    = var.stack-name
+    Scenario = var.scenario-name
+  }
+}
+
+resource "aws_api_gateway_resource" "cg_resource" {
+  rest_api_id = aws_api_gateway_rest_api.cg_api.id
+  parent_id   = aws_api_gateway_rest_api.cg_api.root_resource_id
+  path_part   = "resource-${var.cgid}"
+}
+
+resource "aws_api_gateway_method" "cg_method" {
+  rest_api_id   = aws_api_gateway_rest_api.cg_api.id
+  resource_id   = aws_api_gateway_resource.cg_resource.id
+  http_method   = "GET"
+  authorization = "NONE"
+  api_key_required = true
+}
+
+resource "aws_api_gateway_integration" "cg_integration" {
+  rest_api_id = aws_api_gateway_rest_api.cg_api.id
+  resource_id = aws_api_gateway_resource.cg_resource.id
+  http_method = aws_api_gateway_method.cg_method.http_method
+  type        = "MOCK"
+  request_templates = {
+    "application/json" = "{\"statusCode\": 200}"
+  }
+}
+
+resource "aws_api_gateway_method_response" "cg_method_response" {
+  rest_api_id = aws_api_gateway_rest_api.cg_api.id
+  resource_id = aws_api_gateway_resource.cg_resource.id
+  http_method = aws_api_gateway_method.cg_method.http_method
+  status_code = "200"
+}
+
+resource "aws_api_gateway_integration_response" "cg_integration_response" {
```

Comment:
I was able to skip the SNS part of the scenario with just enumerating the IAM perms

```bash
aws apigateway get-integration --rest-api-id rylal4l89l --resource-id swuayb --http-method GET
```

```json
{
    "type": "MOCK",
    "requestTemplates": {
        "application/json": "{\"statusCode\": 200}"
    },
    "passthroughBehavior": "WHEN_NO_MATCH",
    "timeoutInMillis": 29000,
    "cacheNamespace": "swuayb",
    "cacheKeyParameters": [],
    "integrationResponses": {
        "200": {
            "statusCode": "200",
            "responseTemplates": {
                "application/json": "    {\n      \"message\": \"Access granted.\",\n      \"user_data\": {\n        \"user_id\": \"1337\",\n        \"username\": \"SuperAdmin\",\n        \"email\": \"SuperAdmin@notarealemail.com\",\n        \"password\": \"p@ssw0rd123\"\n      },\n      \"final_flag\": \"FLAG{SNS_S3cr3t$_ar3_FUN}\"\n    }\n"
            }
        }
    }
}
```

Reply:
Good find! This is similar to an unintended path @DaveYesland found. I fixed this one by updating the IAM permissions for an explicit deny. I then tested the attack path and verified this unintended way no longer works.